Tutorials/intro-to-octave/configuration.py

```diff
@@ -56,127 +40,151 @@
             "intermediate_frequency": IF,
             "operations": {
                 "cw": "const",
+                "readout": "readout_pulse",
             },
             "digitalInputs": {
                 "switch": {
-                    "port": ("con1", 4),
+                    "port": (con, 7),
                     "delay": 136,
                     "buffer": 0,
                 },
             },
         },
         "qe5": {
-            "mixInputs": {
-                "I": ("con1", 9),
-                "Q": ("con1", 10),
-                "lo_frequency": LO,
-                "mixer": "octave_octave1_5",  # a fixed name, do not change.
-            },
+            'RF_inputs': {"port": (octave, 5)},
             "intermediate_frequency": IF,
             "operations": {
                 "cw": "const",
+                "readout": "readout_pulse",
             },
             "digitalInputs": {
                 "switch": {
-                    "port": ("con1", 5),
+                    "port": (con, 9),
                     "delay": 136,
                     "buffer": 0,
                 },
             },
         },
+
+    },
+    "octaves": {
+        octave: {
+            "RF_outputs": {
+                1: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",  # can be external or internal. internal is the default
+                    "output_mode": "always_on", # can be: "always_on" / "always_off"/ "triggered" / "triggered_reversed". "always_off" is the default
+                    "gain": 0, # can be in the range [-20 : 0.5 : 20]dB
+                },
+                2: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                3: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                4: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+                5: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal",
+                    "output_mode": "always_on",
+                    "gain": 0,
+                },
+            },
+            "RF_inputs": {
+                1: {
+                    "LO_frequency": LO,
+                    "LO_source": "internal", # internal is the default
+                    "IF_mode_I": "direct",  # can be: "direct" / "mixer" / "envelope" / "off". direct is default
+                    "IF_mode_Q": "direct",
+                },
+                2: {
+                    "LO_frequency": LO,
+                    "LO_source": "external", # external is the default
```

Comment:
Another question, if I want to down-convert the readout signal with a different LO (I play at 6 GHz and I want to down-convert at 12GHz), how can I do it with the new API? (this use-case corresponds to what Scarlino is doing for instance)

Reply:
You can have a different external LO to the down converter... or are you talking about the mixer option?

Reply:
Yeah my question was not clear sorry...
What I mean is under "RF_inputs", we specify the LO, but what does it change? For instance if I have 
```
              1: {
                  "LO_frequency": LO_down,
                  "LO_source": "internal",
              },
```
Then the LO frequency of the down-conversion must be the one given by RF1 no matter the value of LO_down, no?